app/controllers/api/consignment_inquiries_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Api
+  class ConsignmentInquiriesController < RestController
+    before_action :require_trusted_app_without_current_user, only: %i[create]
+
+    def create
+      param! :email, String, required: true
+      param! :name, String, required: true
+      param! :message, String, required: true
+      param! :phone_number, String, required: false
+      param! :gravity_user_id, String, required: false
+      consignment_inquiry = ConsignmentInquiry.create!(consignment_inquiry_params)
+      ConsignmentInquiryService.post_consignment_created_event(consignment_inquiry)
+      render json: consignment_inquiry.to_json, status: :created
+    end
+
+    private
+
+    def require_trusted_app_without_current_user
+      has_trusted_app =
+        current_app.present? &&
+          current_user_roles.include?(:trusted)
+      return if has_trusted_app
+
+      raise ApplicationController::NotAuthorized
+    end
```

Comment:
**suggestion:** We could use the already existing method from [Api::RestControllser](https://github.com/artsy/convection/blob/5dabc41b0ec00f9e0e1a3c46c2471ae38f0d8095/app/controllers/api/rest_controller.rb#L29).

Reply:
The existing method requires that `current_user.blank` == true. However for consignment inquiries we want to be able to create whether we have current_user or not.

Reply:
Hmm, what do you think about renaming the method to `require_trusted_app_and_user` and move it to `Api::RestControllser`? Like this, we can reuse it in the future and it is clear what is requred.

Reply:
👍 . But the right name should be `require_trusted_app` since we don't need to reference user in it, but unfortunately that name is already in use and references current_user. 
How about I move this method as it is to Api::RestController?

Reply:
sounds good to me 👍